### PR TITLE
Removing all batching code from parquet writer

### LINKED ISSUE #13440: [FEA] Remove all batch processing code in parquet writer
As batch processing is no longer supported (https://github.com/rapidsai/cudf/pull/13438), we should remove all the relevant code from parquet writer.

Note that we can also optimize memory usage to some extent, such as keeping only the memory blocks that is needed for writing to sink (either compressed or uncompressed), instead of keeping all.

### PR DESCRIPTION
**Pull request by mhaseeb123**
## Description
Fixes #13440. Removing the manually disabled batching code from parquet writer.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by davidwendt — APPROVED** [review]
LGTM

File: cpp/src/io/parquet/writer_impl.cu [hunk 1]
```diff
@@ -2417,18 +2371,18 @@ void writer::impl::write_parquet_data_to_sink(
   host_2dspan<EncColumnChunk const> chunks,
   host_span<size_t const> global_rowgroup_base,
   host_span<int const> first_rg_in_part,
-  host_span<size_type const> batch_list,
+  size_type const num_rowgroups,
   host_span<int const> rg_to_part,
   host_span<uint8_t> bounce_buffer)
 {
   _agg_meta              = std::move(updated_agg_meta);
   auto const num_columns = chunks.size().second;
```
Comment (etseidl) [inline cpp/src/io/parquet/writer_impl.cu:2379]:
```suggestion
  host_span<int const> rg_to_part,
  host_span<uint8_t> bounce_buffer)
{
  _agg_meta                = std::move(updated_agg_meta);
  auto const num_rowgroups = chunks.size().first;
  auto const num_columns   = chunks.size().second;
```

File: cpp/src/io/parquet/writer_impl.cu [hunk 2]
```diff
@@ -2472,10 +2426,10 @@ void writer::impl::write_parquet_data_to_sink(
...
 
     // add column and offset indexes to metadata
-    for (auto b = 0, r = 0; b < static_cast<size_type>(batch_list.size()); b++) {
-      auto const rnext   = r + batch_list[b];
-      auto curr_page_idx = chunks[r][0].first_page;
-      for (; r < rnext; r++) {
+    if (num_rowgroups != 0) {
+      auto const rnext   = num_rowgroups;
+      auto curr_page_idx = chunks[0][0].first_page;
+      for (auto r = 0; r < rnext; r++) {
```
Comment (etseidl) [inline cpp/src/io/parquet/writer_impl.cu:2432]:
```suggestion
      auto curr_page_idx = chunks[0][0].first_page;
      for (auto r = 0; r < num_rowgroups; r++) {
```

**Review by etseidl — COMMENTED** [review]
Looks really good, just one small suggestion.

File: cpp/src/io/parquet/writer_impl.cu (hunk 2 above)
Comment (mhaseeb123) [reply cpp/src/io/parquet/writer_impl.cu:2432]: Thanks for the suggestions. I tried to remove all duplicate variables but looks like `rnext` slipped under my radar. Let me commit all these suggestions.

Comment (mhaseeb123) [reply cpp/src/io/parquet/writer_impl.cu]: Committed changes. Thanks again!

**Review by ttnghia — APPROVED** [review]

**Pull request merged**